### ISSUE TITLE
Enhancement: `With[{x=0},{y = x}, ...]` support

### ISSUE THREAD

**Issue by Masterxilo** [halirutan/Mathematica-IntelliJ-Plugin#54]
An undocumented feature of `With` is that it allows for constructs like

``` Mathematica
With[{x = 1}, {y = x + 1}, {x, y, x + y}]
```

where it is possible to define variables like `y` using earlier defined variables. However neither this nor the official IDE or _Mathematica_ support this. In _Mathematica_ this construct evaluates correctly but the syntax-highlighting suggests an error:

![Mathematica graphics](http://i.stack.imgur.com/ESRXP.png)

**Comment by halirutan** [comment]
The feature itself is really nice. From the viewpoint of the IDE it is not so nice, because currently, I can handle `Module`, `Block` and `With` with the same algorithm. There are other localization constructs like `Compile`, `Table`, `Manipulate`, etc. and all of them have different ways how where the variables that these functions localize are defined.

As far as I can see, it should not be too hard, to support this, but there is a crucial point against it: it does not work in versions earlier than 10.3. 

My current scheme of support different language levels is non-existent and I have already thought about making the *Mathematica* version adjustable. One problem I see is that we really need to switch the parser because things like `<| ... |>` are really extensions of the language. A possible solution might be to always use the latest language and parse all code with the syntax of e.g. version 11. Afterwards, I can implement code inspections that mark different things as *not supported in your version*.

In summary, I think we should wait until I have time to support different *language versions* until we support this.

**Comment by szhorvat** [comment]
I wonder which version added this. In 10.0 it doesn't work. In 10.3 it does.

**Comment by halirutan** [comment]
@szhorvat I'm also somewhat careless about having every single version installed these days. 4GB download size of *Mathematica* 11 says it all. I have 8.0, 9.0, 10.3.1 and 11.0 currently on my Linux box.

**Comment by halirutan** [comment]
@Masterxilo Quick update:

> In summary, I think we should wait until I have time to support different language versions until we support this.

This is implemented now and if I'm going to work on #59, I'll guess I can implement this as well.

**Comment by halirutan** [comment]
I guess I'm including this in the next big release. I have rewritten the core part that handles how local variables in Module, Table, etc are found and this allows me to include this functionality:

![with](http://i.imgur.com/IhR6YpG.png)

I need to write an error inspection that marks this as invalid as soon as someone uses an old Mathematica version.

**Comment by halirutan** [comment]
This is available in the beta of 3.0
